demos/showcase/alloy-showcase-webapp/src/test/java/com/liferay/faces/test/showcase/TesterBase.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (c) 2000-2015 Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+package com.liferay.faces.test.showcase;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+
+
+/**
+ * @author  Kyle Stiemann
+ */
+public class TesterBase {
+
+	private static final String baseURL;
+	private static final boolean runningWithMavenSurefirePlugin = Boolean.valueOf(System.getProperty(
+				"runningWithMavenSurefirePlugin"));
+	/* package-private */ static String host = System.getProperty("integrationHost", "localhost");
+	/* package-private */ static String port = System.getProperty("integrationPort", "8080");
+	/* package-private */ static String context = System.getProperty("integrationContext",
+			"/alloy-showcase-webapp-3.0.0-SNAPSHOT/web/guest/showcase/-/component/alloy/");
+	/* package-private */ static String container = System.getProperty("integrationContainer", "tomcat");
+	/* package-private */ static String signInContext = System.getProperty("integrationSignInContext", "");
+	protected static final String signInURL;
+	protected static final String testBaseURL;
+
+	static {
+		baseURL = "http://" + host + ":" + port;
+		signInURL = baseURL + signInContext;
+		testBaseURL = baseURL + context;
+	}
+
+	/**
+	 * {@link TestSuiteListener#testRunFinished()} is used to shut down the browser/webDriver when the tests are run
+	 * with the maven-surefire-plugin. However, {@link TestSuiteListener#testRunFinished()} is not called when the tests
+	 * are not run with the maven-surefire-plugin (i.e. when the tests are run from an IDE). So when the tests are run
+	 * from an IDE, it is necessary to shutdown the browser after each test class is run.
+	 */
+	@AfterClass
+	public static void tearDown() {
+
+		if (!runningWithMavenSurefirePlugin) {
+
+			// When the browser is phantomjs or chrome, WebDriver.close() does not quit the browser (like it is
+			// supposed to
+			// https://selenium.googlecode.com/svn/trunk/docs/api/java/org/openqa/selenium/WebDriver.html#quit%28%29),
+			// so we use WebDriver.quit() instead.
+			Browser.getInstance().quit();
+		}
+	}
+
+	/* package-private */ static void signIn() {
+
+		if ("liferay".equals(container)) {
+			// TODO
+		}
+		else if ("pluto".equals(container)) {
+			// TODO
+		}
```

Comment:
@ngriffin7a, I haven't completed the sign in method yet because I'm not sure if the showcase is ready to run in Liferay or Pluto.